cnf-certification-test/performance/suite.go

```diff
@@ -232,30 +233,50 @@ func testSchedulingPolicyInCPUPool(env *provider.TestEnvironment,
 	testhelper.AddTestResultReason(compliantContainersPids, nonCompliantContainersPids, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 
+const noProcessFoundErrMsg = "No such process"
+
 func testRtAppsNoExecProbes(env *provider.TestEnvironment, cuts []*provider.Container) {
 	var compliantObjects []*testhelper.ReportObject
 	var nonCompliantObjects []*testhelper.ReportObject
 	for _, cut := range cuts {
+		if !cut.HasExecProbes() {
+			compliantObjects = append(compliantObjects, testhelper.NewContainerReportObject(cut.Namespace, cut.Podname, cut.Name, "Container does not define exec probes", true))
+			continue
+		}
+
 		processes, err := crclient.GetContainerProcesses(cut, env)
 		if err != nil {
 			tnf.ClaimFilePrintf("Could not determine the processes pids for container %s, err: %v", cut, err)
 			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewContainerReportObject(cut.Namespace, cut.Podname, cut.Name, "Could not determine the processes pids for container", false))
 			break
 		}
+
+		allProcessesCompliant := true
 		for _, p := range processes {
 			schedPolicy, _, err := scheduling.GetProcessCPUScheduling(p.Pid, cut)
 			if err != nil {
+				// If the process does not exist anymore it means that it has finished since the time the process list
+				// was retrieved. In this case, just ignore the error and continue processing the rest of the processes.
+				if strings.Contains(err.Error(), noProcessFoundErrMsg) {
```

Comment:
I would still like to see these in the failure reason. In the reason we can state that the process disappeared before we could do the check. This is not a usual case (as far as I understand) and could qualify for an exception in my opinion.

Reply:
I'm not sure. I don't think we should include the container into the non-compliant category just because by chance we checked one process at the time it was about to disappear. Especially because it is unlikely that this type of processes have a RT policy. Usually RT processes run in busy-loop.

Reply:
"Usually RT processes run in busy-loop" -> it could be because the process repetitively fails to start or other failure scenarios. For certification, I would rather have more understanding of the issue. Then maybe we could add it in the reason for success? Since this condition will lead to a successful certification and at the same time we can know about it.

Reply:
We could skip (by matching command line) processes spawned by exec probes from consideration in this test. Looks like they were the source of the short lived processes.

Reply:
Done.